### PR TITLE
Feature/detect failure case

### LINKED ISSUE #2756: Improve initial web page for the web driver
### Is your feature request related to a problem? Please describe:
Currently their is no user feedback during the initialization of the web driver. It would be good to provide user feedback during the progress of the initialization.

### Is it possible to construct a solution with the existing API?
There should be no need for any new API.

### Describe the solution you'd like to see:
The index.html provided by fyne could:
- detect that the browser does not have javascript support, display warning if not
- detect that the browser support WebGL, display warning if not
- detect that the browser support webassembly, fallback to gopherjs backend
- let the user know that the webassembly is being downloaded

### PR DESCRIPTION
**Pull request by Bluebugs**
### Description:
This is an initial patch that improve detection of failure case when starting a web application. There should be discussion and follow up PR to actually improve the visual of this.

Fixes #2756 

### Checklist:

- [X] Lint and formatter run with no errors.
- [X] Tests all pass.

### CODE REVIEW

File: cmd/fyne_demo/FyneApp.toml [hunk 1]
```diff
@@ -2,4 +2,4 @@
   Icon = "../../theme/icons/fyne.png"
   Name = "Fyne Demo"
   ID = "io.fyne.demo"
-  Build = 6
```
Comment (andydotxyz) [inline cmd/fyne_demo/FyneApp.toml:5]: As this is not releasing a new version of these apps I don't think the build should be bumped...

**Review by andydotxyz — COMMENTED** [review]
Seems like a good change - but I don't think the metadata update belongs

File: cmd/fyne_demo/FyneApp.toml (hunk 1 above)
Comment (Bluebugs) [reply cmd/fyne_demo/FyneApp.toml:5]: Oh indeed.

**Review by andydotxyz — APPROVED** [review]
Should result in better debug info!

**Pull request merged**